# Camera intrinsics
fast_calib:
  ros__parameters:
    # mid360
    #fx: 522.123514287681
    #fy: 522.275153384482
    #cx: 773.466430504725
    #cy: 534.053165700174
    #k1: 0.00324949759262203
    #k2: -0.0171040538369167
    #p1: 0.000669657443377146
    #p2: -0.000350205468789575

    fx: 968.6909
    fy: 965.2046
    cx: 309.268
    cy: 231.423

    k1: -0.01838
    k2: 0.95088
    p1: -0.00339
    p2: -0.00436
    #k3: -3.4511

    # Calibration target parameters
    marker_size: 0.18 # ArUco marker size (our test data uses 0.16m; adjust to match your marker size)
    delta_width_qr_center: 0.61 # Half the distance between the centers of two markers in the horizontal direction
    delta_height_qr_center: 0.41 # Half the distance between the centers of two markers in the vertical direction
    delta_width_circles: 0.3 # Distance between the centers of two circles in the horizontal direction
    delta_height_circles: 0.3 # Distance between the centers of two circles in the vertical direction
    circle_radius: 0.12 # Radius of the circle

    # Distance filter
    x_min: 1.5
    x_max: 3.0
    y_min: -1.5
    y_max: 2.0
    z_min: -0.5 # avia and mid360: -0.5 ouster: -0.8
    z_max: 2.0

    lidar_topic: "/livox/lidar" # /ouster/points /livox/points
    bag_path: "/home/pi-mid70/rosbags/rosbag_calibration_002"
    image_path: "/home/pi-mid70/rosbags/calibration_img_002.png"


    # Output
    output_path: "/home/pi-mid70/ws_livox/calibration_images"

    # Add minimum detected markers parameter
    min_detected_markers: 3
